File: Livrili-main/Backend/MarketplaceBackend-main/src/main/java/com/example/marketplace/model/SellerOrder.java
package com.example.marketplace.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class SellerOrder extends Order{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private Account seller;
    @OneToMany(mappedBy = "order")
    private List<OrderItem> orderItems;
    @ManyToOne
    @JsonIgnore
    private BuyerOrder buyerOrder;
}

File: Livrili-main/Frontend/src/app/shared/models/SellerOrder.ts
import { Account } from "./Account";
import { BuyerOrder } from "./BuyerOrder";
import { OrderItem } from "./OrderItem";

export interface SellerOrder  {
    id?: number;
    totalAmount?: number;
    status?: string;
    date?: Date;
    
    seller?: Account;
    orderItems?: OrderItem[];
    buyerOrder?:BuyerOrder
}